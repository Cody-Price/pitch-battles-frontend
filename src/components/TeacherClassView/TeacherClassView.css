.class-row {
	display: grid;
	grid-template-columns: 200px 1fr 1fr 1fr 1fr 1fr 1fr;
}

.class-table {
	width: 700px;
	margin: auto;
}

