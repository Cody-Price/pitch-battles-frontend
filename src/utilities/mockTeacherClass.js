export const mockTeacherClass = {
  data: {
    id: "5",
    type: "teacher_dashboard",
    attributes: {
      name: "6th Grade Orchestra",
      class_key: "K5s5qCQfHTYYldIhDcmGQg",
      teacher: {
        data: {
          id: "6",
          type: "bare_user",
          attributes: {
            email: "relasine@gmail.com",
            first_name: "Kevin",
            last_name: "Simpson",
            avatar: 1,
            role: "teacher",
            level_one_fastest_time: 0,
            level_two_fastest_time: 0,
            level_three_fastest_time: 0,
            level_four_fastest_time: 0,
            total_games_played: 0
          }
        }
      },
      students: {
        data: [
          {
            id: "3",
            type: "user",
            attributes: {
              email: "dmeskis@gmail.com",
              first_name: "Dylan",
              last_name: "Meskis",
              avatar: 11,
              role: "student",
              level_one_fastest_time: 13402,
              level_two_fastest_time: 24216,
              level_three_fastest_time: 1234134,
              level_four_fastest_time: 0,
              total_games_played: 8,
              games: {
                data: [
                  {
                    id: "11",
                    type: "game",
                    attributes: {
                      total_duration: 33333,
                      level_one_duration: 11111,
                      level_two_duration: 22222,
                      level_three_duration: null,
                      level_four_duration: null,
                      level_one_perfect: true,
                      level_two_perfect: true,
                      level_three_perfect: true,
                      level_four_perfect: true,
                      all_perfect: true
                    }
                  },
                  {
                    id: "12",
                    type: "game",
                    attributes: {
                      total_duration: 33333,
                      level_one_duration: 11111,
                      level_two_duration: 22222,
                      level_three_duration: null,
                      level_four_duration: null,
                      level_one_perfect: true,
                      level_two_perfect: false,
                      level_three_perfect: false,
                      level_four_perfect: false,
                      all_perfect: false
                    }
                  }
                ]
              },
              badges: {
                data: [
                  {
                    id: "1",
                    type: "badge",
                    attributes: {
                      name: "play 5 games",
                      description: "Play five games."
                    }
                  },
                  {
                    id: "2",
                    type: "badge",
                    attributes: {
                      name: "level one: completed",
                      description: "Complete level one."
                    }
                  }
                ]
              },
              class: {
                data: {
                  id: "5",
                  type: "klass",
                  attributes: {
                    name: "6th Grade Orchestra",
                    class_key: "K5s5qCQfHTYYldIhDcmGQg"
                  }
                }
              }
            }
          }
        ]
      },
      level_one_fastest_time: {
        score: 9401,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      },
      level_two_fastest_time: {
        score: 15412,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      },
      level_three_fastest_time: {
        score: 21217,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      },
      level_four_fastest_time: {
        score: 27623,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      },
      overall_fastest_time: {
        score: 33333,
        user: {
          data: [
            {
              id: "3",
              type: "bare_user",
              attributes: {
                email: "dmeskis@gmail.com",
                first_name: "Dylan",
                last_name: "Meskis",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 13402,
                level_two_fastest_time: 24216,
                level_three_fastest_time: 1234134,
                level_four_fastest_time: 0,
                total_games_played: 8
              }
            }
          ]
        }
      },
      most_games: {
        games_played: 24,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      },
      most_badges: {
        badges: 12,
        user: {
          data: [
            {
              id: "1",
              type: "bare_user",
              attributes: {
                email: "simpsonkevinjohn@gmail.com",
                first_name: "Chet",
                last_name: "Manly",
                avatar: 11,
                role: "student",
                level_one_fastest_time: 9401,
                level_two_fastest_time: 15412,
                level_three_fastest_time: 21217,
                level_four_fastest_time: 27623,
                total_games_played: 24
              }
            }
          ]
        }
      }
    }
  }
};
