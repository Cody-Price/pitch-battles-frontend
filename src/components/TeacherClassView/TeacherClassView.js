import React, { Component } from "react";
import "./TeacherClassView.css";
import LoadingAnimation from "../LoadingAnimation/LoadingAnimation";
import { teacherSpecificClassFetch } from "../../utilities/fetchCalls";
import StudentRow from "../StudentRow/StudentRow.js";

class TeacherClassView extends Component {
  constructor() {
    super();
    this.state = {
      klass: undefined,
      students: [],
      loading: true
    };
  }

  componentDidMount() {
    this.classFetch();
  }

  classFetch = async () => {
    try {
      const klass = await teacherSpecificClassFetch(
        this.props.id,
        this.props.webToken
      );

      this.setState({
        klass: klass.data.attributes,
        students: klass.data.attributes.students.data,
        loading: false,
        error: false
      });
    } catch (error) {
      this.setState({
        error: true,
        loading: false
      });
    }
  };

  render() {
    const students = this.state.students.map(student => {
      return (
        <StudentRow
          student={student}
          key={student.id}
          selectStudent={this.props.selectStudent}
        />
      );
    });
    return (
      <section className="teacher-class-view">
        {this.state.loading && (
          <LoadingAnimation message="loading..." animals="cat" />
        )}
        {this.state.error && (
          <LoadingAnimation
            message="- server error - click the home icon to go back"
            animals="duck"
          />
        )}
        {this.state.klass && (
          <div className="loaded-teacher-class-view">
            <h3 className="class-name">{this.state.klass.name}</h3>
            <section className="class-table">
              <article className="class-row">
                <p>Name</p>
                <p>Total</p>
                <p>Level 1</p>
                <p>Level 2</p>
                <p>Level 3</p>
                <p>Level 4</p>
                <p>Delete</p>
              </article>
              {students}
            </section>
          </div>
        )}
      </section>
    );
  }
}

export default TeacherClassView;
