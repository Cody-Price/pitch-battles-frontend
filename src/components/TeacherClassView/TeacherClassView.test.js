import React from "react";
import ReactDOM from "react-dom";
import TeacherClassView from "./TeacherClassView";
import Enzyme, { shallow } from "enzyme";
import { mockTeacherClass } from '../../utilities/mockTeacherClass.js';

const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });
jest.mock("../../utilities/fetchCalls.js");

describe("TeacherClassView", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TeacherClassView />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  describe('classFetch', () => {

  	it('should set state', async () => {
  		await wrapper.instance().classFetch()
  		const expected = mockTeacherClass;

  		wrapper.instance().classFetch();

      	expect(wrapper.state().klass).toEqual(expected.data.attributes);
  	})
  })
});
