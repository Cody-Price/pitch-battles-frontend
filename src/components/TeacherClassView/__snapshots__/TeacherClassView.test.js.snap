// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherClassView should match the snapshot 1`] = `
<section
  className="teacher-class-view"
>
  <div
    className="loaded-teacher-class-view"
  >
    <h3
      className="class-name"
    >
      6th Grade Orchestra
    </h3>
    <section
      className="class-table"
    >
      <article
        className="class-row"
      >
        <p>
          Name
        </p>
        <p>
          Total
        </p>
        <p>
          Level 1
        </p>
        <p>
          Level 2
        </p>
        <p>
          Level 3
        </p>
        <p>
          Level 4
        </p>
        <p>
          Delete
        </p>
      </article>
      <StudentRow
        key="3"
        refresh={[Function]}
        student={
          Object {
            "attributes": Object {
              "avatar": 11,
              "badges": Object {
                "data": Array [
                  Object {
                    "attributes": Object {
                      "description": "Play five games.",
                      "name": "play 5 games",
                    },
                    "id": "1",
                    "type": "badge",
                  },
                  Object {
                    "attributes": Object {
                      "description": "Complete level one.",
                      "name": "level one: completed",
                    },
                    "id": "2",
                    "type": "badge",
                  },
                ],
              },
              "class": Object {
                "data": Object {
                  "attributes": Object {
                    "class_key": "K5s5qCQfHTYYldIhDcmGQg",
                    "name": "6th Grade Orchestra",
                  },
                  "id": "5",
                  "type": "klass",
                },
              },
              "email": "dmeskis@gmail.com",
              "first_name": "Dylan",
              "games": Object {
                "data": Array [
                  Object {
                    "attributes": Object {
                      "all_perfect": true,
                      "level_four_duration": null,
                      "level_four_perfect": true,
                      "level_one_duration": 11111,
                      "level_one_perfect": true,
                      "level_three_duration": null,
                      "level_three_perfect": true,
                      "level_two_duration": 22222,
                      "level_two_perfect": true,
                      "total_duration": 33333,
                    },
                    "id": "11",
                    "type": "game",
                  },
                  Object {
                    "attributes": Object {
                      "all_perfect": false,
                      "level_four_duration": null,
                      "level_four_perfect": false,
                      "level_one_duration": 11111,
                      "level_one_perfect": true,
                      "level_three_duration": null,
                      "level_three_perfect": false,
                      "level_two_duration": 22222,
                      "level_two_perfect": false,
                      "total_duration": 33333,
                    },
                    "id": "12",
                    "type": "game",
                  },
                ],
              },
              "last_name": "Meskis",
              "level_four_fastest_time": 0,
              "level_one_fastest_time": 13402,
              "level_three_fastest_time": 1234134,
              "level_two_fastest_time": 24216,
              "role": "student",
              "total_games_played": 8,
            },
            "id": "3",
            "type": "user",
          }
        }
      />
    </section>
  </div>
</section>
`;
